pyproject.toml

```
@@ -165,7 +167,7 @@ addopts = [
     "-vvv",
     "--import-mode=importlib",
     "--disable-warnings",
-    "--ignore=tests/*",
+    "--ignore=data_safe_haven/resources/*",
```

```suggestion
    "--ignore=data_safe_haven/resources/*",
    "--omit=tests/*",
    "--omit=data_safe_haven/resources/*",
```

I think the idea here was to omit tests from the coverage.

---

Maybe I'm misunderstanding how `coverage` and `pytest` interact. We already have `coverage run --omit=tests/*` in the `test:test` command - I think these options are to `pytest` itself. What I'd like to do is exclude `data_safe_haven/resources/*` from the list of places to look for tests. Do you know the correct syntax for that?

---

> We already have coverage run --omit=tests/* in the test:test command

Ah, maybe we should move it to the options here for consistency.
My comment was assuming removing the ignore for tests would mean they are no longer omitted. I didn't see the other omit argument.

AFAIK, ignore stops pytest from searching matching paths/files for test files (like `test*.py` and omit prevents pytest-cov from counting files in the coverage stats.
I think before everything in `tests/` was being included in the coverage and making it seem lower than it really was.

I think also, if you run `hatch run test:test` you will only run tests in `tests/` as that is the default argument. So "tests" in `data_safe_haven/resources/` should only be found if you run something like `hatch run test:test ./`

---

I'm pretty sure that `--omit` is an argument to `coverage` (see [here](https://coverage.readthedocs.io/en/7.5.4/cmd.html#cmd-run)) and it's telling `coverage` to ignore coverage in the `tests` directory.

Similarly the `pytest` documentation [says that](https://docs.pytest.org/en/7.1.x/example/pythoncollection.html?highlight=ignore) `--ignore` is the correct way to indicate directories that you don't want to be searched when looking for tests.

I think this change is correct.

---

Yes, I think we are saying the same thing.

I'm suggesting that we have all ignores and omits in the same place,

```toml
addopts=[
    ...,
    "--ignore=data_safe_haven/resources/*",
    "--omit=tests/*",
    "--omit=data_safe_haven/resources/*",
]
```

Rather than some in the test script and some in the toml options.

---

Right, but the things in `addopts` are options for `pytest`. We need `--omit=tests/*` to be an option to `coverage` (so that coverage of the test folder doesn't count towards overall stats) and `--ignore=data_safe_haven/resources/*` to be an option to `pytest` (so that we don't try to run the smoke tests with pytest).

---

🤦

---

I'll push changes to the correct place.